imcsdk/apis/admin/ntp.py

```diff
@@ -0,0 +1,175 @@
+    return mo
+
+
+def _set_ntp_servers(mo, ntp_servers):
+    args = {}
+    for server in ntp_servers:
+        key = "ntp_server" + str(server["id"])
+        args[key] = server["ip"]
+    mo.set_prop_multiple(**args)
+
+
+def ntp_enable(handle, ntp_servers=[]):
+    """
+    Enable NTP
+
+    Args:
+        handle (ImcHandle)
+        ntp_servers (list): List of dictionaries of the type
+                            [{"id": 1, "ip": "192.168.1.1"},
+                             {"id": 2, "ip": "192.168.1.2"}]
+                            Upto 4 ntp servers can be specified.
+
+    Returns:
+        CommNtpProvider object
+
+    Example:
+        ntp_enable(handle,
+                   ntp_servers = [{"id": 1, "ip": "192.168.1.1"},
+                                  {"id": 2, "ip": "192.168.1.2"}]
+    """
+
+    mo = _get_ntp_mo()
+    mo.ntp_enable = "yes"
+
+    _set_ntp_servers(mo, ntp_servers)
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def ntp_disable(handle):
+    """
+    Disable NTP
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        CommNtpProvider object
+    """
+
+    mo = _get_ntp_mo()
+    mo.ntp_enable = "no"
+
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def ntp_servers_modify(handle, ntp_servers=[]):
+    """
+    Modify the configured NTP servers
+    Args:
+        handle (ImcHandle)
+        ntp_servers (list): List of dictionaries of the type
+                            [{"id": 1, "ip": "192.168.1.1"},
+                             {"id": 2, "ip": "192.168.1.2"}]
+                            Upto 4 ntp servers can be specified.
+
+    Returns:
+        CommNtpProvider object
+
+    Example:
+        ntp_servers_modify(handle,
+                           ntp_servers = [{"id": 1, "ip": "192.168.1.1"},
+                                          {"id": 2, "ip": "192.168.1.2"},
+                                          {"id": 3, "ip": ""}]
+    """
+
+    # While sending the modified list of servers, it is imperative to send
+    # ntp_enable property in the request.
+    # Hence, query the MO and reassign the same value to ntp_enable
+    mo = _get_ntp_mo(handle, from_server=True)
+    mo.ntp_enable = mo.ntp_enable
+    _set_ntp_servers(mo, ntp_servers)
+
+    handle.set_mo(mo)
+    return mo
+
+
+def is_ntp_enabled(handle):
+    """
+    Check if NTP is enabled
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        bool
+    """
+
+    mo = _get_ntp_mo(handle, from_server=True)
+    return (mo.ntp_enable.lower() in ["true", "yes"])
+
+
+def _is_invalid_value(value):
+    return value in ["", None]
+
+
+def _check_ntp_server_match(ntp_mo, mo):
+    for prop in ["ntp_server1", "ntp_server2",
+                 "ntp_server3", "ntp_server4"]:
+        configured_value = getattr(ntp_mo, prop)
+        in_value = getattr(mo, prop)
+
+        if _is_invalid_value(configured_value) and \
+                _is_invalid_value(in_value):
+            continue
```

Comment:
is the invalid check required? wouldn't the `configured_value != in_value` cover all cases?